src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -339,11 +339,13 @@ private Settings createJwtAuthenticatorSettings(Settings settings) {
         settingsBuilder.put(jwtSettings);
 
         if (jwtSettings.get("roles_key") == null && settings.get("roles_key") != null) {
-            settingsBuilder.put("roles_key", "roles");
+            settingsBuilder.put("roles_key", settings.get("roles_key"));
```

Comment:
What was the issue here again? 
Looks like this settings are used to extract details from a JWT token. 

What was the behavior before and what it is now?

Reply:
By default when we're using SAML we are settings both the JWT roles_key and subject_key and the SAML roles_key and subject_key. JWT roles_key and subject_key are not used while trying to decode the SAML Response. 

One issue is when we are logging SAML errors we are using JWT roles_key and subject_key which is causing confusion. That is why I am explicitly setting the JWT roles_key value to the SAML roles_key for all Saml related calls . (https://github.com/opendistro-for-elasticsearch/security/blob/master/src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java)